### PR TITLE
Update allow list settings access

### PR DESCRIPTION
**Pull request by dkmyta**
Fixes #
- Updates the allow list logic so that the settings are accessible regardless of the state of the related security features.
- Collapses Allow and Block lists in the Jetpack security settings when disabled or unpopulated.
- General Protect UI adjustments as per design.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [x] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

- https://github.com/Automattic/jetpack-scan-team/issues/1359

## Does this pull request change what data or activity we track or use?

- No

## Testing instructions:

* Checkout this branch for Protect and Jetpack on a fresh JN instance using the beta tester
* Run through the various scenarios outlined in the screenshots below
* Ensure the proposed fixes/updates apply
* Ensure that no regressions are introduced

## Screenshots

Jetpack block list enabled:
Screen Shot on 2024-08-14 at 15-01-36

Jetpack block list disabled:
Screen Shot on 2024-08-14 at 14-58-21

Jetpack block list disabled (list populated):
Screen Shot on 2024-08-14 at 15-02-14

Jetpack allow list enabled:
Screen Shot on 2024-08-14 at 15-35-46

Jetpack allow list disabled:
Screen Shot on 2024-08-14 at 15-36-17

Jetpack allow list disabled (list populated):
Screen Shot on 2024-08-14 at 15-36-56

Jetpack Allow list settings always accessible (when all sec features inactive):
   - Supported:
Screen Shot on 2024-08-14 at 15-48-35
   - Unsupported:
Screen Shot on 2024-08-14 at 15-49-42

Protect allow/block list enabled:
Screen Shot on 2024-08-14 at 15-53-28

Protect allow/block list disabled:
Screen Shot on 2024-08-14 at 15-54-22

Protect allow/block list disabled (list populated):
Screen Shot on 2024-08-14 at 15-54-58

Protect Allow list settings always accessible (when all sec features inactive):
   - Supported:
Screen Shot on 2024-08-14 at 15-51-47
   - Unsupported:
Screen Shot on 2024-08-14 at 15-50-55

General adjustment to Protect Firewall UI as per @ilonagl's proposal:
Screen Shot on 2024-08-14 at 16-46-55

### CODE REVIEW

**Comment by ilonagl** [comment]
It looks great! Thanks for doing the changes so quickly. 👏

**Comment by ilonagl** [comment]
I wonder if we could make the buttons bigger? Instead of the small ones (28px) use the default ones (40px) ones?

**Comment by dkmyta** [comment]
>I wonder if we could make the buttons bigger?

If you are referring to the secondary buttons for savings the IP lists and adding your own IP to the allow list, we use a smaller button to maintain consistency/unification with the Jetpack settings. All buttons within that security settings page, and throughout various settings in the core plugin use the `small` variant. Open to changing this approach of course if y'all agree, tagging in @nateweller for his opinion.

File: projects/plugins/jetpack/_inc/client/security/allowList.jsx [hunk 1]
```diff
@@ -137,7 +134,7 @@ const AllowList = class extends Component {
 				onSubmit={ this.onSubmit }
 				hideButton={ true }
 			>
-				{ ( isWafActive || isProtectActive ) && <QueryWafSettings /> }
+				{ <QueryWafSettings /> }
```
Comment (nateweller) [inline projects/plugins/jetpack/_inc/client/security/allowList.jsx]:
```suggestion
				<QueryWafSettings />
```
Very minor

**Review by nateweller — DISMISSED** [review]
Works well! We could follow up with some adjustments to the "save list" button sizes if we like. Since the buttons are only displayed when the lists are enabled, having them larger doesn't seem problematic to me or anything :+1:

**Review by nateweller — APPROVED** [review]

**Pull request merged**